### PR TITLE
Adding fix for edge cases in Space function

### PR DESCRIPTION
**Pull request by tanyagupta17**
### Description

Currently for string function SPACE() we only handle positive integers and throws error for input 0 or negative number. This PR will handle those edge cases and when a negative number is passed we will return NULL and when 0 is passed we will return an empty string.  Additionally, this commit adds handling when input integer is greater than 8000.

Signed-off-by: Tanya Gupta [tanyagp@amazon.com](mailto:tanyagp@amazon.com)

### Issues Resolved

BABEL-4811

### Test Scenarios Covered ###
* **Use case based -** YES

* **Boundary conditions -** YES

* **Arbitrary inputs -** YES

* **Negative test cases -** YES

* **Minor version upgrade tests -** YES

* **Major version upgrade tests -** YES

* **Performance tests -** NO

* **Tooling impact -** NO

* **Client tests -** NO

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 1]
```diff
@@ -1520,7 +1520,13 @@ CREATE OR REPLACE FUNCTION sys.space(IN number INTEGER, OUT result SYS.VARCHAR)
 -- sys.varchar has default length of 1, so we have to pass in 'number' to be the
 -- type modifier.
 BEGIN
-	EXECUTE pg_catalog.format(E'SELECT repeat(\' \', %s)::SYS.VARCHAR(%s)', number, number) INTO result;
+    IF number < 0 THEN
+        result := NULL;
+    ELSEIF number = 0 THEN
+        result := '';
```
Comment (rohit01010) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql]: `PG_CATALOG.repeat()` function returns empty string when number is 0, hence above condition is redundant.

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 2]
```diff
@@ -1520,7 +1520,13 @@ CREATE OR REPLACE FUNCTION sys.space(IN number INTEGER, OUT result SYS.VARCHAR)
 -- sys.varchar has default length of 1, so we have to pass in 'number' to be the
 -- type modifier.
```
Comment (rohit01010) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql]: For functions defined in Postgres with return type `sys.VARCHAR`, the default type modifier here would be `MAX`. Hence this comment is incorrect, we can remove this.

File: test/JDBC/upgrade/13_4/schedule [hunk 3]
```diff
@@ -234,3 +234,4 @@ babel-4517
 BABEL-4641-before-16_3-or-15_7-or-14_12
 BABEL-4863-before-16_3-or-15_7-or-14_12
 BABEL-4869
+space
```
Comment (rohit01010) [inline test/JDBC/upgrade/13_4/schedule:246]: NIT: Add new line at the end of schedule file. Same applicable for all other schedule file as well.

**Review by rohit01010 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 2 above)
Comment (tanyagupta17) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql]: Done

File: test/JDBC/upgrade/13_4/schedule (hunk 3 above)
Comment (tanyagupta17) [reply test/JDBC/upgrade/13_4/schedule:246]: Done

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 1 above)
Comment (tanyagupta17) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql]: Done

**Review by rohit01010 — APPROVED** [review]

**Review by shah-nirmit — APPROVED** [review]

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 4]
```diff
@@ -1517,10 +1517,12 @@ STRICT
...
 
 CREATE OR REPLACE FUNCTION sys.space(IN number INTEGER, OUT result SYS.VARCHAR) AS $$
--- sys.varchar has default length of 1, so we have to pass in 'number' to be the
--- type modifier.
 BEGIN
-	EXECUTE pg_catalog.format(E'SELECT repeat(\' \', %s)::SYS.VARCHAR(%s)', number, number) INTO result;
+    IF number < 0 THEN
+        result := NULL;
+    ELSE
+        result := PG_CATALOG.repeat(' ',number);
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql]: will it be able to handle INT_MAX?

Comment (tanyagupta17) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql]: As per our offline discussion:
TSQL has a limitation of 8000 character spaces for SPACE function. Added the same support now.

**Review by rohit01010 — APPROVED** [review]

**Review by shah-nirmit — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]
